From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Sat, 6 Feb 2021 13:19:31 +0100
Subject: [PATCH] (KigPaper) Better Async Chunk API


diff --git a/src/main/java/org/bukkit/World.java b/src/main/java/org/bukkit/World.java
index 325d65a37dc5e38bf64cb88b936d2a258be182ab..11cc6cca948394d1a0668b46abef07039286480f 100644
--- a/src/main/java/org/bukkit/World.java
+++ b/src/main/java/org/bukkit/World.java
@@ -7,6 +7,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
 
 import org.bukkit.block.Biome;
 import org.bukkit.block.Block;
@@ -130,12 +131,21 @@ public interface World extends PluginMessageRecipient, Metadatable {
     public Chunk getChunkAt(Block block);
 
     // PaperSpigot start - Async chunk load API
+    @Deprecated
     public static interface ChunkLoadCallback {
         public void onLoad(Chunk chunk);
     }
+    @Deprecated
     public void getChunkAtAsync(int x, int z, ChunkLoadCallback cb);
+    @Deprecated
     public void getChunkAtAsync(Location location, ChunkLoadCallback cb);
+    @Deprecated
     public void getChunkAtAsync(Block block, ChunkLoadCallback cb);
+    // KigPaper start - better Java 8 API
+    CompletableFuture<Chunk> getChunkAtAsync(int x, int z);
+    CompletableFuture<Chunk> getChunkAtAsync(Location location);
+    CompletableFuture<Chunk> getChunkAtAsync(Block block);
+    // KigPaper end
     // PaperSpigot end
 
     /**
