From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Fri, 23 Apr 2021 18:19:55 +0200
Subject: [PATCH] (KigPaper) Add setType without lighting update API


diff --git a/src/main/java/org/bukkit/block/Block.java b/src/main/java/org/bukkit/block/Block.java
index 235c15bdfed7a593d408a5b4103aa857824b29f8..ef50d523fa65f66aab62c75b0fdf4a30318ab2dd 100644
--- a/src/main/java/org/bukkit/block/Block.java
+++ b/src/main/java/org/bukkit/block/Block.java
@@ -192,7 +192,9 @@ public interface Block extends Metadatable {
      * @param type Material to change this block to
      * @param applyPhysics False to cancel physics on the changed block.
      */
-    void setType(Material type, boolean applyPhysics);
+    default void setType(Material type, boolean applyPhysics) {
+        setType(type, applyPhysics, true); // KigPaper - add light param
+    }
 
     /**
      * Sets the type-id of this block
@@ -213,7 +215,9 @@ public interface Block extends Metadatable {
      * @deprecated Magic value
      */
     @Deprecated
-    boolean setTypeId(int type, boolean applyPhysics);
+    default boolean setTypeId(int type, boolean applyPhysics) {
+        return setTypeId(type, applyPhysics, true); // KigPaper - light param
+    }
 
     /**
      * Sets the type-id of this block
@@ -225,7 +229,9 @@ public interface Block extends Metadatable {
      * @deprecated Magic value
      */
     @Deprecated
-    boolean setTypeIdAndData(int type, byte data, boolean applyPhysics);
+    default boolean setTypeIdAndData(int type, byte data, boolean applyPhysics) {
+        return setTypeIdAndData(type, data, applyPhysics, true); // KigPaper - light param
+    }
 
     /**
      * Gets the face relation of this block compared to the given block.
@@ -390,4 +396,38 @@ public interface Block extends Metadatable {
      */
     Collection<ItemStack> getDrops(ItemStack tool);
 
+    // KigPaper start
+    /**
+     * Sets the type-id of this block
+     *
+     * @param type Type-Id to change this block to
+     * @param data The data value to change this block to
+     * @param applyPhysics False to cancel physics on the changed block
+     * @param updateLight whether to update light in the chunk
+     * @return whether the block was changed
+     * @deprecated Magic value
+     */
+    @Deprecated
+    boolean setTypeIdAndData(int type, byte data, boolean applyPhysics, boolean updateLight);
+
+    /**
+     * Sets the type of this block
+     *
+     * @param type Material to change this block to
+     * @param applyPhysics False to cancel physics on the changed block.
+     * @param updateLight whether to update light in the chunk
+     */
+    void setType(Material type, boolean applyPhysics, boolean updateLight);
+
+    /**
+     * Sets the type-id of this block
+     *
+     * @param type Type-Id to change this block to
+     * @param applyPhysics False to cancel physics on the changed block.
+     * @param updateLight whether to update light in the chunk
+     * @return whether the block was changed
+     * @deprecated Magic value
+     */
+    boolean setTypeId(int type, boolean applyPhysics, boolean updateLight);
+    // KigPaper end
 }
