From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Mon, 25 Jan 2021 17:36:41 +0100
Subject: [PATCH] (KigPaper) Fix Entity and Command Block memory leaks


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 7fb08af407dcacfbe37222dfd66064ffc3dd6963..0a8a885dc9b03edbd8ce85ccae78727e230cef90 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -217,19 +217,17 @@ public interface Entity extends Metadatable, CommandSender {
     public void setFallDistance(float distance);
 
     /**
-     * Record the last {@link EntityDamageEvent} inflicted on this entity
-     *
-     * @param event a {@link EntityDamageEvent}
+     * Does nothing.
+     * @deprecated This is never used and only leads to memory leaks.
      */
+    @Deprecated
     public void setLastDamageCause(EntityDamageEvent event);
 
     /**
-     * Retrieve the last {@link EntityDamageEvent} inflicted on this entity.
-     * This event may have been cancelled.
-     *
-     * @return the last known {@link EntityDamageEvent} or null if hitherto
-     *     unharmed
+     * Always returns null.
+     * @deprecated This is never used and only leads to memory leaks.
      */
+    @Deprecated // KigPaper - deprecate
     public EntityDamageEvent getLastDamageCause();
 
     /**
