From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: BuildTools <LinsaFTW@users.noreply.github.com>
Date: Tue, 3 Nov 2020 16:19:20 -0300
Subject: [PATCH] (FlamePaper) Return last slot by default


diff --git a/src/main/java/net/minecraft/server/Container.java b/src/main/java/net/minecraft/server/Container.java
index 7fc495792c1caf81511d8197b05e2d0b5ddb71b7..943f068fce399c3ce9c756ebccc1e58777a2c5f8 100644
--- a/src/main/java/net/minecraft/server/Container.java
+++ b/src/main/java/net/minecraft/server/Container.java
@@ -106,6 +106,12 @@ public abstract class Container {
     }
 
     public Slot getSlot(int i) {
+        final int lastIndex = this.c.size() - 1;
+
+        if (i > lastIndex) {
+            i = lastIndex;
+        }
+
         return (Slot) this.c.get(i);
     }
 
