From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "kfian294ma4@gmail.com" <kfian294ma4@gmail.com>
Date: Sat, 19 Dec 2020 18:50:09 +0000
Subject: [PATCH] (IonSpigot) Lag Compensated Pearls


diff --git a/src/main/java/me/suicidalkids/ion/IonWorldConfig.java b/src/main/java/me/suicidalkids/ion/IonWorldConfig.java
index f256d71c9a570ce51b9f50a70e60ea86a8794ea6..22f3378b641a9eaa1833cbd6597cfac02938e3d3 100644
--- a/src/main/java/me/suicidalkids/ion/IonWorldConfig.java
+++ b/src/main/java/me/suicidalkids/ion/IonWorldConfig.java
@@ -118,4 +118,9 @@ public class IonWorldConfig {
         lagCompensatedPotions = getBoolean("potions.lag-compensated", true);
     }
 
+    public boolean lagCompensatedPearls;
+    private void LagCompensatedPearls() {
+        lagCompensatedPearls = getBoolean("pearls.lag-compensated", true);
+    }
+
 }
diff --git a/src/main/java/net/minecraft/server/EntityEnderPearl.java b/src/main/java/net/minecraft/server/EntityEnderPearl.java
index 4817e187e9291affe6675367353644137b7ae566..a1b61f9d94feb075e5d99549eef9a59b7f6f34e2 100644
--- a/src/main/java/net/minecraft/server/EntityEnderPearl.java
+++ b/src/main/java/net/minecraft/server/EntityEnderPearl.java
@@ -21,6 +21,12 @@ public class EntityEnderPearl extends EntityProjectile {
     public EntityEnderPearl(World world, EntityLiving entityliving) {
         super(world, entityliving);
         this.c = entityliving;
+        // IonSpigot start - Lag Compensated Pearls
+        if (entityliving instanceof EntityPlayer && world.ionConfig.lagCompensatedPearls) {
+            ((EntityPlayer) entityliving).lagCompensatedTicking.add(this);
+            compensated = true;
+        }
+        // IonSpigot end
     }
 
     protected void a(MovingObjectPosition movingobjectposition) {
@@ -104,7 +110,15 @@ public class EntityEnderPearl extends EntityProjectile {
 
     }
 
+    // IonSpigot start - Lag Compensated Pearls
+    @Override
     public void t_() {
+        if (!compensated) {
+            tick();
+        }
+    }
+    public void tick() {
+    // IonSpigot end
         EntityLiving entityliving = this.getShooter();
 
         if (entityliving != null && entityliving instanceof EntityHuman && !entityliving.isAlive()) {
