From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Thu, 11 Feb 2021 17:38:57 +0100
Subject: [PATCH] (KigPaper) Fix NPE from 0138 and logging from 0139


diff --git a/src/main/java/net/minecraft/server/EntityInsentient.java b/src/main/java/net/minecraft/server/EntityInsentient.java
index b351add9a8292b8f82237bf0bd748b3f504b39c9..3b806b235b02ccbfe1289ba20fba9292a8ac91da 100644
--- a/src/main/java/net/minecraft/server/EntityInsentient.java
+++ b/src/main/java/net/minecraft/server/EntityInsentient.java
@@ -86,7 +86,7 @@ public abstract class EntityInsentient extends EntityLiving {
     }
 
     public EntityLiving getGoalTarget() {
-        return this.goalTarget.get(); // KigPaper - WeakReference
+        return this.goalTarget == null ? null : this.goalTarget.get(); // KigPaper - WeakReference
     }
 
     public void setGoalTarget(EntityLiving entityliving) {
