From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Fri, 5 Mar 2021 15:16:21 +0100
Subject: [PATCH] Paper-0321 Cleanup allocated favicon ByteBuf


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 5a8a22ab29cc313921e4b2fdca874925070c8564..32c3c4d954f07aa9c5a6dd7d348c18dee4d3359f 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -673,20 +673,24 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
 
         if (file.isFile()) {
             ByteBuf bytebuf = Unpooled.buffer();
-
+            ByteBuf faviconBuffer = null; // KigPaper
             try {
                 BufferedImage bufferedimage = ImageIO.read(file);
 
                 Validate.validState(bufferedimage.getWidth() == 64, "Must be 64 pixels wide", new Object[0]);
                 Validate.validState(bufferedimage.getHeight() == 64, "Must be 64 pixels high", new Object[0]);
                 ImageIO.write(bufferedimage, "PNG", new ByteBufOutputStream(bytebuf));
-                ByteBuf bytebuf1 = Base64.encode(bytebuf);
+                //ByteBuf bytebuf1 = Base64.encode(bytebuf); // KigPaper
+                faviconBuffer = Base64.encode(bytebuf);
 
-                serverping.setFavicon("data:image/png;base64," + bytebuf1.toString(Charsets.UTF_8));
+                serverping.setFavicon("data:image/png;base64," + faviconBuffer.toString(Charsets.UTF_8));
             } catch (Exception exception) {
                 MinecraftServer.LOGGER.error("Couldn\'t load server icon", exception);
             } finally {
                 bytebuf.release();
+                // KigPaper start
+                if(faviconBuffer != null) faviconBuffer.release();
+                // KigPaper end
             }
         }
 
