From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Mon, 26 Apr 2021 09:50:16 +0200
Subject: [PATCH] (KigPaper) Kick after receiving invalid creative action
 packet


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index acb35e8278bc56c45c3c7ee86a752bf48885326e..baa2247592ac4fbfcd715f8a2d396937653180af 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1844,6 +1844,8 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                     entityitem.j();
                 }
             }
+        } else {
+            disconnect("Not in creative mode"); // KigPaper
         }
 
     }
