From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Sun, 4 Jul 2021 17:48:09 +0200
Subject: [PATCH] (KigPaper) Don't calculate initial light if not requested


diff --git a/src/main/java/net/minecraft/server/Chunk.java b/src/main/java/net/minecraft/server/Chunk.java
index 223a8aa445b82d0e231695e424e27f1a84c9fe41..3ea3d5337a19ee7ad60ee048b067573a049533f9 100644
--- a/src/main/java/net/minecraft/server/Chunk.java
+++ b/src/main/java/net/minecraft/server/Chunk.java
@@ -611,7 +611,7 @@ public class Chunk {
             if (chunksection.b(i, j & 15, k) != block) {
                 return null;
             } else {
-                if (flag) {
+                if (updateLight && flag) {
                     this.initLighting();
                 } else if (updateLight) { // KigPaper - conditional light update
                     int j1 = block.p();
