From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Mon, 26 Apr 2021 18:28:33 +0200
Subject: [PATCH] (KigPaper) NBT no-op reader for block place packet


diff --git a/src/main/java/net/minecraft/server/PacketPlayInBlockPlace.java b/src/main/java/net/minecraft/server/PacketPlayInBlockPlace.java
index 02d8e07f89b817b830ddf495f0ce789f7780c340..b305ee0f317efded63005965f09e49704a72251d 100644
--- a/src/main/java/net/minecraft/server/PacketPlayInBlockPlace.java
+++ b/src/main/java/net/minecraft/server/PacketPlayInBlockPlace.java
@@ -1,5 +1,7 @@
 package net.minecraft.server;
 
+import io.netty.handler.codec.DecoderException;
+
 import java.io.IOException;
 
 public class PacketPlayInBlockPlace implements Packet<PacketListenerPlayIn> {
@@ -33,7 +35,12 @@ public class PacketPlayInBlockPlace implements Packet<PacketListenerPlayIn> {
         timestamp = System.currentTimeMillis(); // CraftBukkit
         this.b = packetdataserializer.c();
         this.c = packetdataserializer.readUnsignedByte();
-        this.d = packetdataserializer.i();
+        // KigPaper start - don't parse itemstack
+        // this.d = packetdataserializer.i();
+        // Consume everything and leave 3 bytes at the end
+        if (packetdataserializer.readableBytes() < 3) throw new DecoderException("Expected 3 facing bytes");
+        packetdataserializer.skipBytes(packetdataserializer.readableBytes() - 3);
+        // KigPaper end
         this.e = (float) packetdataserializer.readUnsignedByte() / 16.0F;
         this.f = (float) packetdataserializer.readUnsignedByte() / 16.0F;
         this.g = (float) packetdataserializer.readUnsignedByte() / 16.0F;
