From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <roccodev.business@gmail.com>
Date: Fri, 7 Aug 2020 23:45:58 +0200
Subject: [PATCH] (KigPaper) Make the permission message configurable


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 3f3331d8d4ab343359cea34430c6afd3372b5199..8d7373f075488047b0a4387b404c4d9792762880 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1874,4 +1874,11 @@ public final class CraftServer implements Server {
     {
         return spigot;
     }
+
+    // KigPaper start
+    @Override
+    public String getPermissionMessage() {
+        return configuration.getString("settings.permission-message", ChatColor.RED + "I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.");
+    }
+    // KigPaper end
 }
