From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <roccodev.business@gmail.com>
Date: Fri, 7 Aug 2020 23:45:58 +0200
Subject: [PATCH] (KigPaper) Make the permission message configurable


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 71e5d04da3373db13e4004cf3422ccc269bb1d5d..c4e886fd4b0cc79964ec5aa9d1a98c6b80c32d32 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1875,4 +1875,11 @@ public final class CraftServer implements Server {
     {
         return spigot;
     }
+
+    // KigPaper start
+    @Override
+    public String getPermissionMessage() {
+        return configuration.getString("settings.permission-message", ChatColor.RED + "I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.");
+    }
+    // KigPaper end
 }
