From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Sun, 5 Aug 2018 02:08:01 -0700
Subject: [PATCH] (SportPaper) Fix BlockFromToEvents firing when flow is
 obstructed


diff --git a/src/main/java/net/minecraft/server/BlockFlowing.java b/src/main/java/net/minecraft/server/BlockFlowing.java
index 23130ba963a7fb0024cee47a2a44006e08d06353..b1168dfef77af4816b0ce36bce5fec48a4cbf9a9 100644
--- a/src/main/java/net/minecraft/server/BlockFlowing.java
+++ b/src/main/java/net/minecraft/server/BlockFlowing.java
@@ -143,13 +143,15 @@ public class BlockFlowing extends BlockFluids {
                 EnumDirection enumdirection1 = (EnumDirection) iterator1.next();
 
                 // CraftBukkit start
-                BlockFromToEvent event = new BlockFromToEvent(source, org.bukkit.craftbukkit.block.CraftBlock.notchToBlockFace(enumdirection1));
-                if (server != null) {
-                    server.getPluginManager().callEvent(event);
-                }
+                if(this.h(world, blockposition.shift(enumdirection1), world.getType(blockposition.shift(enumdirection1)))) {
+                    BlockFromToEvent event = new BlockFromToEvent(source, org.bukkit.craftbukkit.block.CraftBlock.notchToBlockFace(enumdirection1));
+                    if (server != null) {
+                        server.getPluginManager().callEvent(event);
+                    }
 
-                if (!event.isCancelled()) {
-                    this.flow(world, blockposition.shift(enumdirection1), world.getType(blockposition.shift(enumdirection1)), k);
+                    if (!event.isCancelled()) {
+                        this.flow(world, blockposition.shift(enumdirection1), world.getType(blockposition.shift(enumdirection1)), k);
+                    }
                 }
                 // CraftBukkit end
             }
